packages/compass-explain-plan/src/components/explain-tree/explain-tree.tsx

```diff
@@ -1,91 +1,95 @@
 
-  useEffect(() => {
-    if (!svgRef.current || !links || !nodes) return;
-    const svgElement = svgRef.current;
+  const highlightsHeight =
+    Object.keys(node.highlights).length * highlightFieldHeight;
 
-    // Right angle links between nodes
-    const elbow = (d: {
-      source: { x: number; y: number; x_size: number };
-      target: { y: number; x: number; key: string };
-    }): string => {
-      return `M${d.source.x + d.source.x_size / 2},${d.source.y}
-        V${d.target.y - STAGE_CARD_PROPERTIES.VERTICAL_PADDING / 2}
-        H${d.target.x + STAGE_CARD_PROPERTIES.DEFAULT_CARD_WIDTH / 2}
-        V${d.target.y}`;
-    };
-    const svg = d3.select(svgElement);
+  const notShardHeight = defaultCardHeight + highlightsHeight;
+  const height = node.isShard ? shardCardHeight : notShardHeight;
 
-    // Links are svg elements
-    const svgLinks = svg
-      .selectAll('path')
-      .data(links, (d) => d.target.key)
-      .attr('d', elbow);
+  return [defaultCardWidth, height];
+};
 
-    svgLinks
-      .enter()
-      .append('path')
-      .style({
-        fill: 'none',
-        stroke: darkMode ? palette.gray.base : palette.gray.light2,
-        'stroke-width': '6px',
-      })
-      .attr('d', elbow);
+const getNodeKey = (node: ExplainTreeNodeData) => node.id;
 
-    svgLinks.exit().remove();
+const ExplainTree: React.FunctionComponent<ExplainTreeProps> = ({
+  executionStats,
+}) => {
+  const darkMode = useDarkMode();
+  const [detailsOpen, setDetailsOpen] = useState<string | null>(null);
 
-    return () => {
-      // make sure to clean up the elements from the previous hook execution
-      d3.select(svgElement).selectAll('*').remove();
-    };
-  }, [nodes, links, darkMode]);
+  const root = useMemo(
+    () => executionStatsToTreeData(executionStats),
+    [executionStats]
+  );
 
-  const getStages = () => {
-    return map(nodes, (stage) => <ExplainStage {...stage} />);
-  };
+  if (!root) return null;
 
   return (
-    <div
+    <TreeLayout<ExplainTreeNodeData>
       data-testid="explain-tree"
+      data={root}
+      getNodeSize={getNodeSize}
+      getNodeKey={getNodeKey}
+      linkColor={darkMode ? palette.gray.base : palette.gray.light2}
+      linkWidth={6}
+      horizontalSpacing={spacing[6]}
+      verticalSpacing={spacing[6]}
       className={explainTreeStyles}
-      style={{ height, width }}
     >
-      {getStages()}
-      <svg width="100%" height="100%" ref={svgRef}></svg>
-    </div>
+      {(node) => {
+        const key = getNodeKey(node);
+        return (
+          <div
+            style={{
+              position: 'relative',
+              zIndex: detailsOpen === key ? 2 : 1,
+            }}
+          >
+            <ExplainTreeStage
+              detailsOpen={detailsOpen === key}
+              toggleDetails={() => {
+                detailsOpen === key
+                  ? setDetailsOpen(null)
+                  : setDetailsOpen(key);
+              }}
```

Comment:
good point, I should have brought this up, but the change is intentional. 

It is a bit of an assumption based on the fact that the previous version ends up with the same UX by accident since in most cases the detail panel just hides the other detail buttons. We chatted a bit with Claudia about this but we didn't find a use case where having 2 panels open at the same time is a must.